### PR TITLE
[DO NOT MERGE] Revert routed_scaling_factor fusion — Nemotron CI debug for release

### PR DESCRIPTION
**Pull request by b8zhong**
**Note: since this PR is still failing, it seems not related to this revert**

Debug-only PR for the Nemotron-3-Super NVFP4 + MTP NaN (`verify: target model logits`) on `release/v0.5.13`.

Reverts **only** the routed_scaling_factor fusion introduced in #26733, restoring the original `final_hidden_states *= self.routed_scaling_factor` in `NemotronHMoE.forward`:
- `nemotron_h.py`: drop `routed_scaling_factor` from the experts call; `topk` back to `routed_scaling_factor=1.0` (no `apply_routed_scaling_factor_on_output`); restore the FP16-overflow scaling block in `forward`.
- `modelopt_quant.py` / `unquant.py`: remove the added `output.mul_(routed_scaling_factor)` blocks.

Unrelated #26733 perf changes (router `torch.mm`, `relu2`, mamba `output=None`, gated-RMSNorm PDL) are left intact, to isolate whether the scaling fusion is implicated.

**Do not merge.**

---
### CI States

Latest PR Test (Base): :x: [Run #27281408240](https://github.com/sgl-project/sglang/actions/runs/27281408240)
Latest PR Test (Extra): :x: [Run #27281407755](https://github.com/sgl-project/sglang/actions/runs/27281407755)

### CODE REVIEW

**Comment by b8zhong** [comment]
/rerun-test test/registered/8-gpu-models/test_nvidia_nemotron_3_super_nightly.py test/registered/4-gpu-models/test_nvidia_nemotron_3_super_nvfp4.py
